app/routes.py

```diff
@@ -1,2 +1,239 @@
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from datetime import datetime 
+#what comes from the "from" is the package name
+#what comes after the import is a variable that matches the name of thing we want to import
+tasks_bp = Blueprint("tasks", __name__,url_prefix=("/tasks"))
+
+goals_bp = Blueprint("goals", __name__, url_prefix=("/goals"))
+
+@goals_bp.route("",methods=["POST", "GET"])
+def handle_goal():
+      if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return{
+                "details": "Invalid data"
+            },400
+
+        new_goal = Goal (
+            title=request_body["title"]
+        )
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return {
+            "goal": {
+                "id":new_goal.goal_id,
+                "title":new_goal.title
+            }
+        }, 201
+        
+      elif request.method == "GET":
+        sorting_goals= request.args.get('sort') 
+        list = None
+        if sorting_goals== "desc":
+            list = Goal.query.order_by(Goal.title.desc()) # descending method
+        elif sorting_goals == "asc":
+            list = Goal.query.order_by(Goal.title.asc()) # ascending method
+        else: 
+            list = Goal.query.all()
+        goals_response = []
+        for goal in list:
+            goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title,
+            })
+
+        return jsonify(goals_response)
+
+@goals_bp.route("/<goal_id>", methods=["GET","PUT","DELETE"])
+def handle_goal_get(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal == None:
+        return ("", 404)
+
+    if request.method == "GET":
+        return {
+            "goal": {
+                "id":goal.goal_id,
+                "title":goal.title,          
+                }
+        }
+    if request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+        
+        db.session.commit()
+
+        return jsonify({
+            "goal":{
+                "id":goal.goal_id,
+                "title":goal.title,
+            }
+        }),200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }),200
+        
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST","GET"])
+def post_tasked_goal(goal_id):
+
+        goal = Goal.query.get(goal_id)
+
+        if goal == None:
+            return (""), 404
+
+        if request.method == "POST":
+            request_body = request.get_json()
+          
+            tasks_instances= []
+            for task_id in request_body["task_ids"]:
+                tasks_instances.append(Task.query.get(task_id))
+            
+            goal.tasks = tasks_instances
```

Comment:
👍 

Assigning to the `tasks` relationship is absolutely a valid way to associate tasks with a goal.  We should also think about whether to interpret POSTing to this endpoint is setting the exact list of tasks that should be a part of the goal, or whether we are adding these tasks to the goal (in addition to anything that was already there). Neither the project description nor the test `test_post_task_ids_to_goal_already_with_goals` are entirely clear on the desired behavior. Which perspective does your implementation take? How could we implement the other approach?